### DISCUSSION TITLE
from nextjs 13.3.0 to 13.3.1-canary.11 socket.io not working

### DISCUSSION THREAD

**Discussion by flipUp02** [vercel/next.js#48510] [Help] [answered]
### Summary

With the same code, socket IO only works on Nextjs version 13.2.4. As for any other version that shows error number 404 for the page http://localhost:3000/socket.io/?EIO= I hope to solve the problem quickly and see the following code: https://github.com/flipUp02/nextjs-socket.io

13 3 1-canary 11
![13 3 1-canary 11](https://user-images.githubusercontent.com/80167896/232636454-a9a9ddd4-2a7b-4e58-a32a-716b129a5004.png)

13.2.4
![13 2 4](https://user-images.githubusercontent.com/80167896/232636520-d54ae6e3-fcb9-4e5a-81f5-88d324588281.png)

please fix this

### Example

https://github.com/flipUp02/nextjs-socket.io

**Comment by flipUp02** [comment]
please fix

**Comment by flipUp02** [comment]
still same problem in nextjs 13.3.1

**Comment by saicharan-gevme** [comment]
Same issue happening for me, needed help on this.
Thanks.

**Comment by jhetuts** [comment]
Bump on this as well!

**Comment by flipUp02** [comment]
We really hope that the nextjs team will solve the problem quickly.

**Comment by Minisaints** [comment]
Bump: I'm also receiving the above 404s. I'm also getting undefined from res.socket in GET handler on route.ts.

**Comment by flipUp02** [comment]
I hope it will be resolved soon

**Comment by flipUp02** [comment]
Still not fix

**Comment by saicharan-gevme** [comment]
@leerob Please check this issue, we are facing issue with socket.io

**Comment by flipUp02** [comment]
@leerob please help

**Comment by saicharan-gevme** [comment]
To be precise the Socket.IO is not working since Next.js 13.2.5-canary.27 version

**Comment by techCodeSB** [comment]
same problem, please fix it

**Comment by flipUp02** [comment]
Unfortunately, the problem has not been resolved yet.

**Comment by vanta240i** [comment]
Please let us know when this is resolved.

**Comment by arye321** [comment]
13.4 still not working

**Reply by spencerchang** [reply]
facing same problem.

**Comment by leerob** [comment, marked as answer]
Could you please open an issue with a reproduction, filling out the issue template? Thank you! We will look into this and get it fixed.

**Reply by saicharan-gevme** [reply]
Thank you very much for responding to my request ❤️
I have created the issue, by explaining about the issue and have also created the example project on code sandbox by reproducing the bug.
Additionally I have also provided a [reference PR](https://github.com/vercel/next.js/pull/47208#issuecomment-1524565169), where this issue could have been created at.

Issue Link - https://github.com/vercel/next.js/issues/49334

I really loved the Next.js so much, would be looking forward to help with anything with regard to this.
And really love to see all your videos on YouTube, I religiously follow them on your YouTube Channel and also on Vercal channel.
Please kindly check this issue as soon as possible.
